Client/Frontend/Browser/New Tab Page/NewTabPageViewController.swift

```diff
@@ -121,30 +126,53 @@ class NewTabPageViewController: UIViewController, Themeable {
             }),
             DuckDuckGoCalloutSectionProvider(profile: profile, action: { [weak self] in
                 self?.delegate?.tappedDuckDuckGoCallout()
-            }),
+            })
         ]
+        
+        if !PrivateBrowsingManager.shared.isPrivateBrowsing {
+            sections.append(
+                BraveTodaySectionProvider(
+                    dataSource: feedDataSource,
+                    actionHandler: handleBraveTodayAction
+                )
+            )
+            layout.braveTodaySection = sections.firstIndex(where: { $0 is BraveTodaySectionProvider })
+        }
+
         collectionView.delegate = self
         collectionView.dataSource = self
         applyTheme(Theme.of(tab))
         
         background.changed = { [weak self] in
             self?.setupBackgroundImage()
         }
+        
+        Preferences.BraveToday.isEnabled.observe(from: self)
+        feedDataSource.observeState(from: self, handleFeedStateChange)
+        NotificationCenter.default.addObserver(self, selector: #selector(checkForUpdatedFeed), name: UIApplication.didBecomeActiveNotification, object: nil)
     }
     
     @available(*, unavailable)
     required init(coder: NSCoder) {
         fatalError()
     }
     
+    deinit {
+        NotificationCenter.default.removeObserver(self)
```

Comment:
This doesn't get called, check for retain cycle somewhere. Now closing a tab doesn't deinitialize it

Reply:
Fixed this, and boy was it a hard one to find 😅

Reply:
good job, I could not find it at first glance too lol